### ISSUE TITLE
[Helm] Adding a serviceMonitor to kube-prometheus values throws helm error

### ISSUE THREAD

**Issue by hobbsh** [coreos/prometheus-operator#1240]
Seems to be an issue with the helm template for serviceMonitors - could it be possible that it is expecting some helm labels for my service, but I didn't deploy my service with helm?

**What did you do?**
Added a service monitor in kube-prometheus values.yml and updated with helm (using Ansible):

**What did you expect to see?**
Expected the serviceMonitor to create on its own

**What did you see instead? Under which circumstances?**
```
{
   "changed":true,
   "cmd":[
      "/usr/local/bin/helm",
      "upgrade",
      "--install",
      "kube-prometheus",
      "coreos/kube-prometheus",
      "--namespace",
      "monitoring",
      "-f",
      "/root/.helm/kube-prometheus-values.yml"
   ],
   "delta":"0:00:01.645034",
   "end":"2018-04-23 02:19:03.529344",
   "msg":"non-zero return code",
   "rc":1,
   "start":"2018-04-23 02:19:01.884310",
   "stderr":"Error: UPGRADE FAILED: render error in \"kube-prometheus/charts/prometheus/templates/servicemonitors.yaml\": template: kube-prometheus/charts/prometheus/templates/servicemonitors.yaml:18:30: executing \"kube-prometheus/charts/prometheus/templates/servicemonitors.yaml\" at <.Values.prometheusLa...>: can't evaluate field prometheusLabelValue in type interface {}",
   "stderr_lines":[
      "Error: UPGRADE FAILED: render error in \"kube-prometheus/charts/prometheus/templates/servicemonitors.yaml\": template: kube-prometheus/charts/prometheus/templates/servicemonitors.yaml:18:30: executing \"kube-prometheus/charts/prometheus/templates/servicemonitors.yaml\" at <.Values.prometheusLa...>: can't evaluate field prometheusLabelValue in type interface {}"
   ]
}
```
**Environment**
Using Ansible (kubespray) to install/upgrade prometheus-operator and kube-prometheus

```
$ sudo helm version
Client: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
```
* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.5", GitCommit:"f01a2bf98249a4db383560443a59bed0c13575df", GitTreeState:"clean", BuildDate:"2018-03-19T15:50:45Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.5", GitCommit:"f01a2bf98249a4db383560443a59bed0c13575df", GitTreeState:"clean", BuildDate:"2018-03-19T15:50:45Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: `kubespray`

* Manifests:
If I `kubectl apply -f lumo-service-monitor.yml -n monitoring` it works fine

```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: web-lumo
  labels:
    k8s-apps: web-lumo
    prometheus: kube-prometheus
spec:
  jobLabel: web-lumo
  selector:
    matchLabel:
      k8s-app: web-lumo
  namespaceSelector:
    matchNames:
    - lumo
  endpoints:
  - port: http
    interval: 30s
```

For comparison, this is the serviceMonitor definition in my `kube-prometheus-values.yml` - some slight differences from trying a bunch of stuff out but nothing alarming IMO. 

```
  serviceMonitors: 
    - name: web-lumo
      labels:
        prometheus: kube-prometheus
      selector:
        matchLabels:
          k8s-app: web-lumo
      endpoints:
        - port: http
          interval: 30s
          path: /metrics
      namespaceSelector:
        any: true
```

**Comment by cspargo** [comment]
i don't have time right now to do a pr for this, but the issue is in helm/prometheus/templates/servicemonitors.yaml

```
prometheus: {{ .Values.prometheusLabelValue | default $releaseName | quote }}
```

should be scoped

```
prometheus: {{ $.Values.prometheusLabelValue | default $releaseName | quote }}
```

**Comment by gianrubio** [comment]
@cspargo I'm gonna fix that, the notation should be `{{ default $releaseName .Values.prometheusLabelValue | quote }}`

**Comment by gianrubio** [comment]
fixed by #1263
